### ISSUE TITLE
specscore.compute_coadd_scores fails for TSNR2_XYZ

### ISSUE THREAD

**Issue by AlexGKim** [desihub/desispec#1193]
This spectra fits file

/global/project/projectdirs/desi/spectro/redux/daily/tiles/80721/20210305/coadd-1-80721-20210305.fits

includes spectra.scores names

TSNR2_ELG_B
TSNR2_BGS_B
TSNR2_QSO_B
TSNR2_LRG_B
TSNR2_ELG_R
TSNR2_BGS_R
TSNR2_QSO_R
TSNR2_LRG_R
TSNR2_ELG_Z
TSNR2_BGS_Z
TSNR2_QSO_Z
TSNR2_LRG_Z
TSNR2_ELG
TSNR2_BGS
TSNR2_QSO
TSNR2_LRG

The latter 4 fail in line 72 of specscore.py, because the band is expected.

**Comment by sbailey** [comment]
Thanks for reporting @AlexGKim.  To facilitate testing and fixing, specifically what command were you running that triggered specscore.py line 72 on this file?

@julienguy @akremin @sbailey we collectively need to be better at testing and announcing data model changes.

**Comment by AlexGKim** [comment]
The error occurred when running `coadd_cameras(spectra)`.

**Comment by sbailey** [comment]
Specifically the error occurs when trying to coadd a coadd:
```
import desispec.io
import desispec.coaddition
spectra = desispec.io.read_spectra('spectra-1-80721-20210305.fits')
coadd0 = desispec.io.read_spectra('coadd-1-80721-20210305.fits')
coadd1 = desispec.coaddition.coadd_cameras(spectra)   # WORKS
coadd2 = desispec.coaddition.coadd_cameras(coadd1)    # FAILS
coadd3 = desispec.coaddition.coadd_cameras(coadd0)    # FAILS
```
Traceback is
```
...
/global/common/software/desi/cori/desiconda/20200801-1.4.0-spec/code/desispec/master/py/desispec/specscore.py in compute_coadd_scores(coadd, specscores, update_coadd)
     70         for colname in specscores.dtype.names:
     71             if colname.startswith('TSNR2_'):
---> 72                 _, targtype, band = colname.split('_')
     73                 scores[colname] = np.zeros(num_targets)
     74                 comments[colname] = f'{targtype} {band} template (S/N)^2'

ValueError: not enough values to unpack (expected 3, got 2)
```

**Comment by michaelJwilson** [comment]
I put a fix in the tsnr-camera-coadd-fix branch that passes all your test cases.  My preferred fix would have been TSNR2_ELG -> TSNR2_ELG_BRZ as the coadded flux is handled and handle the logic accordingly, but I figured you'd prefer the option that didn't require updating all the recent coadds in daily. 

There is a precision issue, in that the BRZ TSNR values in coadd0 are float32, but derived from the float64 BRZ TSNR values in the spectra file.  BRZ are simply the sum of the B, R and Z cols so recalculating the scores for coadd0 yields slightly different results from that actually in the coadd0 file, in some cases, which isn't ideal: 

coadd3.scores['TSNR2_ELG'][1]                                                                                                                                                                                       
Out[1]: 16.729626

In [2]: coadd0.scores['TSNR2_ELG'][1]                                                                                                                                                                                       
Out[2]: 16.729628

They do pass np.allclose for all target types.  I wasn't involved in the tsnr coadding, but I'd imagine I might introduce similar problems as I'm not aware of how to run a set of standard reductions that tests all the use cases.

**Comment by sbailey** [comment]
Thanks @michaelJwilson .  Are there any other unrelated WIP changes in your tsnr-camera-coadd-fix branch that we should be aware of, or could we proceed with a PR for that branch to fix this particular issue?

**Comment by michaelJwilson** [comment]
This was a dedicated branch spun off from master.  I also doubled checked my 4 pt spacing so hopefully you don't need to shout at me about that!  PR is in.

**Comment by sbailey** [comment]
fixed by PR #1197, thanks @michaelJwilson .
